spec/urlHash.spec.js

```diff
@@ -0,0 +1,31 @@
+describe('URL manipulation methods', function() {
+
+  var UrlHash = require('../examples/lib/urlHash')
+  var urlHash;
+  var params = { who: 'me' }
+
+  beforeEach(()=>{
+    urlHash = UrlHash
+
+    spyOn(urlHash,'getUrlHashParameters')
+    spyOn(urlHash,'getUrlHashParameter')
+    spyOn(urlHash,'setUrlHashParameters')
+
+    urlHash.getUrlHashParameters()
+    urlHash.getUrlHashParameter('value')
+    urlHash.setUrlHashParameters(params)
+  })
+
+  it('gets url hash params from window hash', function() {
+    expect(urlHash.getUrlHashParameters).toHaveBeenCalled()
```

Comment:
we can test the getUrlHashParameters() function with an argument and using toHaveBeenCalledWith matcher

Reply:
`getUrlHashParameters` doesn't take arguments, so I didn't test it for argument calls. All the rest do though, so I'll add them in!